runtime/interface.go

```diff
@@ -41,93 +43,110 @@ type Block struct {
-	Log(string) error
-	// EmitEvent is called when an event is emitted by the runtime.
-	EmitEvent(cadence.Event) error
+	UpdateContractCode(address AddressLocation, code []byte, caller Location) (err error)
+	// RemoveContractCode removes the code associated with an account contract.
+	RemoveContractCode(address AddressLocation, caller Location) (err error)
+
+	// Value gets a value for the given key in the storage, owned by the given account.
+	Value(address Address, key []byte, caller Location) (value []byte, err error)
+	// SetValue sets a value for the given key in the storage, owned by the given account.
+	SetValue(address Address, key []byte, value []byte, caller Location) (err error)
 	// ValueExists returns true if the given key exists in the storage, owned by the given account.
-	ValueExists(owner, key []byte) (exists bool, err error)
-	// GenerateUUID is called to generate a UUID.
-	GenerateUUID() (uint64, error)
-	// GetComputationLimit returns the computation limit. A value <= 0 means there is no limit
-	GetComputationLimit() uint64
-	// SetComputationUsed reports the amount of computation used.
-	SetComputationUsed(used uint64) error
-	// DecodeArgument decodes a transaction argument against the given type.
-	DecodeArgument(argument []byte, argumentType cadence.Type) (cadence.Value, error)
-	// GetCurrentBlockHeight returns the current block height.
-	GetCurrentBlockHeight() (uint64, error)
-	// GetBlockAtHeight returns the block at the given height.
-	GetBlockAtHeight(height uint64) (block Block, exists bool, err error)
-	// UnsafeRandom returns a random uint64, where the process of random number derivation is not cryptographically
-	// secure.
-	UnsafeRandom() (uint64, error)
-	// VerifySignature returns true if the given signature was produced by signing the given tag + data
-	// using the given public key, signature algorithm, and hash algorithm.
-	VerifySignature(
-		signature []byte,
-		tag string,
-		signedData []byte,
-		publicKey []byte,
-		signatureAlgorithm string,
-		hashAlgorithm string,
-	) (bool, error)
-	// Hash returns the digest of hashing the given data with using the given hash algorithm
-	Hash(data []byte, hashAlgorithm string) ([]byte, error)
-	// GetStorageUsed gets storage used in bytes by the address at the moment of the function call.
-	GetStorageUsed(address Address) (value uint64, err error)
-	// GetStorageCapacity gets storage capacity in bytes on the address.
-	GetStorageCapacity(address Address) (value uint64, err error)
-}
-
-type HighLevelStorage interface {
-	Interface
+	ValueExists(address Address, key []byte, caller Location) (exists bool, err error)
 
-	// HighLevelStorageEnabled should return true
-	// if the functions of HighLevelStorage should be called,
-	// e.g. SetCadenceValue
-	HighLevelStorageEnabled() bool
+	// StoredKeys returns list of keys and their sizes owned by this account
+	StoredKeys(address Address, caller Location) (keys [][]byte, sizes []uint64, err error)
```

Comment:
I like this one a lot!

Reply:
Yeah, should remove the need for reading registers to compute storage changes

Reply:
This will be very useful, nice!

The might get very large, do we maybe want to have this be some sort of iterator?

Reply:
changed it to an iterator